src/components/app/data/hooks/useAlgoliaSearch.test.tsx

```diff
@@ -0,0 +1,215 @@
+      searchClient: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+      searchIndex: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+    }
+    : {
+      catalogUuidsToCatalogQueryUuids: mockEmptyBaseAlgoliaData.catalogUuidsToCatalogQueryUuids,
+      searchClient: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+      searchIndex: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+    };
+};
+
+const matchQueryFnToURLPattern = ({ enterpriseSlug, queryFnToMatch }) => {
+  const queryFnToURLPattern = [{
+    route: `/${enterpriseSlug}`,
+    queryFn: fetchEnterpriseLearnerDashboard,
+  },
+  {
+    route: `/${enterpriseSlug}/search/`,
+    queryFn: fetchEnterpriseLearnerSearch,
+  },
+  {
+    route: `/${enterpriseSlug}/academies/academy-uuid`,
+    queryFn: fetchEnterpriseLearnerAcademy,
+  },
+  {
+    route: `/${enterpriseSlug}/skills-quiz`,
+    queryFn: fetchEnterpriseLearnerSkillsQuiz,
+  },
+  ];
+  const matchedQueryFnToRoute = queryFnToURLPattern.find(({ queryFn }) => queryFn === queryFnToMatch);
+  return matchedQueryFnToRoute?.route;
+};
+
+describe('useAlgoliaSearch', () => {
+  const Wrapper = ({
+    initialEntries = [], children,
+  }: { initialEntries?: string[] | undefined; children: React.ReactNode }) => (
+    <QueryClientProvider client={queryClient()}>
+      <MemoryRouter initialEntries={initialEntries}>
+        <Routes>
+          <Route path=":enterpriseSlug" element={children} />
+          <Route path=":enterpriseSlug/search" element={children} />
+          <Route path=":enterpriseSlug/academies/:academyUUID" element={children} />
+          <Route path=":enterpriseSlug/skills-quiz" element={children} />
+          <Route path=":enterpriseSlug/unsupported-bff-route" element={children} />
+        </Routes>
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // @ts-ignore
+    mockedUseEnterpriseCustomer.mockReturnValue({
+      data: mockEnterpriseCustomer,
+    });
+    // @ts-ignore
+    mockedUseEnterpriseFeatures.mockReturnValue({
+      data: { catalogQuerySearchFiltersEnabled: true },
+    });
+    (fetchEnterpriseLearnerDashboard as jest.Mock).mockResolvedValue(mockBaseBFFData);
+    (fetchEnterpriseLearnerSearch as jest.Mock).mockResolvedValue(mockBFFSearchData);
+    (fetchEnterpriseLearnerAcademy as jest.Mock).mockResolvedValue(mockBFFAcademyData);
+    (fetchEnterpriseLearnerSkillsQuiz as jest.Mock).mockResolvedValue(mockBFFSkillsQuizData);
+    getConfig.mockReturnValue(APP_CONFIG);
+  });
+
+  it.each([
+    // BFF Disabled Routes
+    ...generateTestPermutations({
+      isMatchedBFFRoute: [false],
+      indexName: [
+        APP_CONFIG.ALGOLIA_INDEX_NAME,
+        APP_CONFIG.ALGOLIA_INDEX_NAME_JOBS,
+      ],
+      isCatalogQueryFiltersEnabled: [false, true],
+    }),
+    ...generateTestPermutations({
```

Comment:
nit: are these permutations distinct from the "// BFF Disabled Routes" mentioned above? Or does this set of permutations need a separate comment, e.g. "// BFF Enabled Routes"